Contents/Code/siteWowGirls.py

```diff
@@ -0,0 +1,95 @@
+import PAsearchSites
+import PAgenres
+import PAactors
+
+def search(results,encodedTitle,title,searchTitle,siteNum,lang,searchByDateActor,searchDate,searchSiteID):
+    if searchSiteID != 9999:
+        siteNum = searchSiteID
+    searchResults = HTML.ElementFromURL(PAsearchSites.getSearchSearchURL(siteNum) + encodedTitle)
+    index = 0
+    for searchResult in searchResults.xpath('//div[@class="videos-list"]'):
+        titleNoFormatting = searchResult.xpath('.//article/a')[0].text_content().strip()
+        curID = searchResult.xpath('.//article/a')[0].get('href').replace('/','_').replace('?','!')
+        # no Date aviable on Site => taking Date from File
+        releaseDate = searchDate
+        score = 100 - Util.LevenshteinDistance(searchTitle.lower(), titleNoFormatting.lower())
+        results.Append(MetadataSearchResult(id = curID + "|" + str(siteNum), name = titleNoFormatting + " [WowPorn/WowGirls] " + releaseDate, score = score, lang = lang))
+        index += 1
+
+    return results
+
+def update(metadata,siteID,movieGenres,movieActors):
+    Log('******UPDATE CALLED*******')
+
+    url = str(metadata.id).split("|")[0].replace('_','/').replace('?','!')
+    detailsPageElements = HTML.ElementFromURL(url)
+    art = []
+    metadata.collections.clear()
+    movieGenres.clearGenres()
+    movieActors.clearActors()
+
+    # Studio
+    metadata.studio = 'WowGirls'
+
+    # Title
+    metadata.title = detailsPageElements.xpath('//div[@class="title-views"]/h1')[0].text_content().strip().rsplit(" ",3)[0]
+
+    # Summary
+    metadata.summary = detailsPageElements.xpath('//div[@class="desc "]/p')[0].text_content().strip()
+
+    #Tagline and Collection(s)
+    tagline = PAsearchSites.getSearchSiteName(siteID).strip()
+    metadata.tagline = tagline
+    metadata.collections.add(tagline)
+
+    # Genres
+    genres = detailsPageElements.xpath('//div[@class="tags-list"]/a[i[@class="fa fa-folder"]]')
+    if len(genres) > 0:
+        for genreLink in genres:
+            genreName = genreLink.text_content().strip().lower()
+            movieGenres.addGenre(genreName)
+    movieGenres.addGenre("Genre")
+
+    # Release Date
+    date = ""
+    if len(date) > 0:
+        date_object = datetime.strptime(date, '%Y-%m-%d')
+        metadata.originally_available_at = date_object
+        metadata.year = metadata.originally_available_at.year
+    
+    # Actors
+    actors = detailsPageElements.xpath('//div[@class="tags-list"]/a[i[@class="fa fa-tag"]]')
+    if len(actors) > 0:
+        if len(actors) == 3:
+            movieGenres.addGenre("Threesome")
+        if len(actors) == 4:
+            movieGenres.addGenre("Foursome")
+        if len(actors) > 4:
+            movieGenres.addGenre("Orgy")
+        for actorLink in actors:
```

Comment:
"Actor Page URL" is a method to scrape an image from the actor's profile page on that specific site. If you don't want to bother with that (as the agent does have alternate means to find profile pics for many actors), just remove all that code and use this:

```
        for actorLink in actors:
            actorName = str(actorLink.text_content().strip())
            actorPhotoURL = ''
            movieActors.addActor(actorName,actorPhotoURL)
```

Your method is trying to telling the agent that the actor's profile page is actually an image file that should be scraped. Therefore, no image is being and no fallback method is being used (thus a blank image for every actor).

Reply:
done